src/extension.ts

```diff
@@ -1,59 +1,53 @@
-                    let r = new vscode.Range(p, p.translate(0, match[0].length + 1))
-                    if (match[1]) {
-                        let uri = vscode.Uri.parse("https://github.com/" + match[1] + "/issues/" + match[2])
-                        res.push(new vscode.DocumentLink(r, uri))
-                    } else if (repo.length > 0) {
-                        let uri = vscode.Uri.parse("https://github.com/" + repo + "/issues/" + match[2])
-                        res.push(new vscode.DocumentLink(r, uri))
-                    }
-                }
-                return res
-            }
+export async function activate(context: vscode.ExtensionContext) {
+    const originRepo = vscode.workspace.rootPath ? await parseRepoName(vscode.workspace.rootPath) : null
+
+    const disposable = vscode.languages.registerDocumentLinkProvider('*', {
+        provideDocumentLinks: function (document: vscode.TextDocument): vscode.DocumentLink[] {
+            return generateIssueLinks(document, originRepo)
         }
+    });
+    context.subscriptions.push(disposable);
+}
 
-        let dispose = vscode.languages.registerDocumentLinkProvider('*', provider);
-        context.subscriptions.push(dispose);
+function generateIssueLinks(document: vscode.TextDocument, originRepo: string | null): vscode.DocumentLink[] {
+    const re = /([a-zA-Z0-9_\-]+\/[a-zA-Z0-9_\-]+)?#(\d+)/g
+    const txt = document.getText()
+    const res: vscode.DocumentLink[] = []
+    let match
+    while ((match = re.exec(txt)) !== null) {
+        const position = document.positionAt(match.index)
+        const range = new vscode.Range(position, position.translate(0, match[0].length))
+        const uri = vscode.Uri.parse(`https://github.com/${match[1] || originRepo}/issues/${match[2]}`)
+        res.push(new vscode.DocumentLink(range, uri))
     }
-    if (vscode.workspace.rootPath !== undefined) {
-        let gconfig = path.join(vscode.workspace.rootPath, '.git', 'config')
-        fs.readFile(gconfig, 'utf-8', (err, data) => {
-            if (err) {
-                linkIssues("")
-            } else {
-                for (let line of data.split('\n')) {
-                    let match = /\s*url\s*=.*github.com\/([^\/]+\/[^\/]+)$/.exec(line)
-                    if (match) {
-                        let rawRepo = match[1]
-                        // Transform "dt/ghlink.git" into "dt/ghlink"
-                        let repo = rawRepo.endsWith('.git') ? rawRepo.substring(0, rawRepo.length - '.git'.length) : rawRepo
-                        linkIssues(repo)
-                        return
-                    }
-                }
-            }
-        })
+    return res
+}
+
+async function parseRepoName(directory: string): Promise<string | null> {
+    try {
+        const pathToConfig = join(directory, '.git', 'config')
+        const config = await parseGitConfig({ path: pathToConfig, expandKeys: true })
+        return urlToRepoName(config.remote.origin.url)
```

Comment:
Hmm... This assumes everyone names their upstream remote `origin`? Previously I think it would work even if you named it something else. What would you think of iterating over the remotes until one has a URL that matches GitHub? Or maybe trying "origin" first and if it doesn't exist, iterating?

Reply:
Good catch! 👍

Let's check `origin` first and then iterate like you said. This covers the case where I have `origin` but it's not the first one in the list for some reason.

I'll fix this later, maybe tomorrow.